src/CADL.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -0,0 +1,446 @@
+        } else if (isInputUnionType(contentTypeParameter.Type)) {
+            const mediaTypeValues =
+                contentTypeParameter.Type.UnionItemTypes.map((item) =>
+                    isInputLiteralType(item) ? item.Value : undefined
+                );
+            if (mediaTypeValues.some((item) => item === undefined)) {
+                throw "Media type of content type should be string.";
+            }
+            mediaTypes.push(...mediaTypeValues);
+        }
+    }
+    const requestMethod = parseHttpRequestMethod(verb);
+    const generateProtocol: boolean = shouldGenerateProtocol(dpgContext, op);
+    const generateConvenience: boolean =
+        requestMethod !== RequestMethod.PATCH &&
+        shouldGenerateConvenient(dpgContext, op);
+
+    /* handle lro */
+    /* handle paging. */
+    let paging: OperationPaging | undefined = undefined;
+    for (const res of operation.responses) {
+        const body = res.responses[0]?.body;
+        if (body?.type) {
+            if (
+                body.type.kind === "Model" &&
+                hasDecorator(body?.type, "$pagedResult")
+            ) {
+                const itemsProperty = Array.from(
+                    body.type.properties.values()
+                ).find((it) => hasDecorator(it, "$items"));
+                const nextLinkProperty = Array.from(
+                    body.type.properties.values()
+                ).find((it) => hasDecorator(it, "$nextLink"));
+                paging = {
+                    NextLinkName: nextLinkProperty?.name,
+                    ItemName: itemsProperty?.name
+                } as OperationPaging;
+            }
+        }
+    }
+    /* TODO: handle lro */
+
+    return {
+        Name: op.name,
+        ResourceName:
+            resourceOperation?.resourceType.name ??
+            getOperationGroupName(program, op, serviceNamespaceType),
+        Summary: summary,
+        Deprecated: getDeprecated(program, op),
+        Description: desc,
+        Parameters: parameters,
+        Responses: responses,
+        HttpMethod: requestMethod,
+        RequestBodyMediaType: typeToBodyMediaType(cadlParameters.body?.type),
+        Uri: uri,
+        Path: fullPath,
+        ExternalDocsUrl: externalDocs?.url,
+        RequestMediaTypes: mediaTypes.length > 0 ? mediaTypes : undefined,
+        BufferResponse: true,
+        LongRunning: loadOperationLongRunning(
+            program,
+            operation,
+            resourceOperation
+        ),
+        Paging: paging,
+        GenerateProtocolMethod: generateProtocol,
+        GenerateConvenienceMethod: generateConvenience
+    } as InputOperation;
+
+    function loadOperationParameter(
+        program: Program,
+        parameter: HttpOperationParameter
+    ): InputParameter {
+        const { type: location, name, param } = parameter;
+        let format = undefined;
+        if (parameter.type !== "path") {
+            format = parameter.format;
+        }
```

Comment:
could we refine this?
```suggestion
        const format = parameter.type === "path" ? undefined : parameter.format;
```
Or we might could change it to just `const format = parameter.format;` because we will never have a `format` defined on a path parameter from the compiler. Even if we have, we will not use it below

Reply:
updated.